infer/src/base/ForkUtils.ml

```diff
@@ -9,10 +9,15 @@ open! IStd
 module L = Logging
 
 let protect ~f x =
-  Epilogues.reset () ;
-  L.reset_formatters () ;
-  ResultsDatabase.new_database_connection () ;
-  (* get different streams of random numbers in each fork, in particular to lessen contention in
-     `Filename.mk_temp` *)
-  Random.self_init () ;
+  if Config.is_child then
+    (* Few things need to be done in fork-exec mode, as the process just went through Infer's
+       initialization phase: most of the steps below (except opening the db) just got done. *)
```

Comment:
Is this change needed for correctness or is it an optimisation? Not sure it's worth the added complexity if the latter.

Reply:
Optimization. It's indeed very minor, will remove.

Reply:
Ah sorry, I was thinking about another change. Actually, there was a problem with both the epilogues and the output streams without that change. OTOH, we introduced this in our "old" fork. In the current version, "child mode" proceeds a bit differently. I will experiment.